packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -160,6 +180,62 @@ export interface ReplicaRegion {
   readonly encryptionKey?: kms.IKey;
 }
 
+/**
+ * An experimental class used to specify an initial secret value for a Secret.
+ * The class wraps a simple string (or JSON representation) in order to provide some safety checks and warnings
+ * about the dangers of using plaintext strings as initial secret seed values via CDK/CloudFormation.
+ */
+export class SecretStringValueBeta1 {
+
+  /**
+   * Creates a `SecretStringValueBeta1` from a plaintext value.
+   * This approach is inherently unsafe, as the secret value will be visible in plaintext in the resulting
+   * CloudFormation template, including in the AWS Console or APIs.
+   * Usage of this method is discouraged, especially for production workloads.
+   */
+  public static fromUnsafePlaintext(secretValue: string) { return new SecretStringValueBeta1(secretValue); }
+
+  /**
+   * Creates a `SecretValueValueBeta1` from a string value coming from a Token.
+   * The intent is to enable creating secrets from references (e.g., `Ref`, `Fn::GetAtt`) from other resources.
+   * This might be the direct output of another Construct, or the output of a Custom Resource.
+   * This method throws if it determines the input is an unsafe plaintext string.
+   *
+   * For example:
+   *     // Creates a new IAM user, access and secret keys, and stores the secret access key in a Secret.
+   *     const user = new iam.User(this, 'User');
+   *     const accessKey = new iam.CfnAccessKey(this, 'AccessKey', { userName: user.userName });
+   *     const secretValue = SecretStringValueBeta1.fromToken(accessKey.attrSecretAccessKey);
+   *     new secretsmanager.Secret(this, 'Secret', {
+   *       secretValueBeta1: secretValue,
+   *     });
+   *
+   * The secret may also be embedded in a string representation of a JSON structure:
+   *     const secretValue = SecretStringValueBeta1.fromToken(JSON.stringify({
+   *       username: user.userName,
+   *       database: 'foo',
+   *       password: accessKey.attrSecretAccessKey
+   *     }));
+   *
+   * Note that the value being a Token does *not* guarantee safety. For example, a Lazy-evaluated string
+   * (e.g., `Lazy.string({ produce: () => 'myInsecurePassword' }))`) is a Token, but as the output is
```

Comment:
We can technically check that (i.e resolve the token and see if it's eventually a Ref or a GetAtt). Resolution can be done inside a wrapping token.

Reply:
Yes. This is a bit complex and/or error-prone, however, which is why I opted for the simplest solution to start.

Simply checking for the presence of `Ref` in the string doesn't guarantee it's a CloudFormation reference (e.g., `passRef1234`). I can try to regex it, but then I have to account for nuances in formatting, especially as this Ref/GetAtt can be embedded inside an arbitrary JSON structure. Something like `/{\s*"?\s*(Ref|Fn::GetAtt)\s*"?\s*:/m` _might_ work. I worry about unintentionally blocking an edge case we haven't thought of this way. WDYT? Is it worth the ugly regex and potentially blocking something if we've missed vs protecting against people passing in a Lazy value?